akka-http-caching/src/main/scala/akka/http/caching/LfuCache.scala

```diff
@@ -111,7 +112,20 @@ private[caching] class LfuCache[K, V](val store: AsyncLoadingCache[K, V]) extend
 
   def apply(key: K, genValue: () ⇒ Future[V]): Future[V] = store.get(key, toJavaMappingFunction[K, V](genValue)).toScala
 
-  def getOrLoad(key: K, loadValue: K ⇒ Future[V]) = store.get(key, toJavaMappingFunction[K, V](loadValue)).toScala
+  def getOrLoad(key: K, loadValue: K ⇒ Future[V]): Future[V] =
+    store.get(key, toJavaMappingFunction[K, V](loadValue)).toScala
+
+  def put(key: K, mayBeValue: Future[V])(implicit ex: ExecutionContext): Future[Unit] = {
+    val cachedBefore = store.synchronous().asMap().containsKey(key)
```

Comment:
I think `store.getIfPresent(key) != null` might be better as it allows for an optimized implementation

Reply:
Done